File: c++/1929-concatenation-of-array.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int N = nums.size();
        vector<int> result(2 * N);

        for(int i = 0; i < N; ++ i) {
            result[i]     = nums[i];
            result[N + i] = nums[i];
        }

        return result;
    }
};

File: dev-rust/1929-concatenation-of-array.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let mut result: Vec<i32> = vec![0; nums.len() * 2];

        for i in 0..nums.len() {
            result[i] = nums[i];
            result[i + nums.len()] = nums[i];
        }    

        return result;
    }
}